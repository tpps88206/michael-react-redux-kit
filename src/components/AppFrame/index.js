import React from 'react';

import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import MenuIcon from '@material-ui/icons/Menu';

const AppFrame = ({ children }) => {
  return (
    <div>
      <AppBar position="static">
        <Toolbar>
          <IconButton edge="start" color="inherit" aria-label="menu">
            <MenuIcon />
          </IconButton>
          <Typography variant="h6">Github Repositories Searching</Typography>
        </Toolbar>
      </AppBar>
      <div>{children}</div>
    </div>
  );
};

export default AppFrame;
