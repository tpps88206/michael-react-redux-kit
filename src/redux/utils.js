import { ofType } from 'redux-observable';

import flatten from 'lodash/flatten';
import fromPairs from 'lodash/fromPairs';
import isEmpty from 'lodash/isEmpty';
import { mergeMap } from 'rxjs/operators';

// there are two type of handler, different in key name:
// - `actionName`: map handler to `actionName`
// - `actionNameEpic`: get full control of epic function
const sliceToEpics = slice => {
  if (!slice || !slice.epics) {
    return [];
  }
  const actionTypes = fromPairs(
    Object.entries(slice)
      .filter(([k]) => k !== 'default' && k !== 'epics')
      .filter(([, v]) => v && v.type)
      .map(([k, v]) => [k, v.type]),
  );
  const epics = Object.entries(slice.epics).map(([k, handler]) => {
    if (k.endsWith('Epic')) {
      return handler;
    }

    return (action$, state$) => {
      return action$.pipe(
        ofType(actionTypes[k]),
        mergeMap(action => handler(action$, state$, action)),
      );
    };
  });

  return epics;
};

export const getEpicsFromSlices = slices => {
  if (isEmpty(slices)) {
    return [];
  }
  return flatten(slices.map(sliceToEpics));
};
