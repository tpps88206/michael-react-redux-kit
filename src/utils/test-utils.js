import React from 'react';
import { Provider } from 'react-redux';

import { create } from 'jss';
import preset from 'jss-preset-default';

import CssBaseline from '@material-ui/core/CssBaseline';
import { createMuiTheme, StylesProvider, ThemeProvider } from '@material-ui/core/styles';
import { render as rtlRender } from '@testing-library/react';

import theme from '@/constants/theme';
import configureStore from '@/redux/configureStore';
import styles from '@/styles';

const render = (ui, { ...renderOptions } = {}) => {
  const muiTheme = createMuiTheme(theme);
  const jss = create({
    plugins: [...preset().plugins],
  });
  const store = configureStore();

  jss.createStyleSheet(styles(muiTheme), { meta: 'global' }).attach();

  const Wrapper = ({ children }) => {
    return (
      <StylesProvider jss={jss}>
        <ThemeProvider theme={muiTheme}>
          <CssBaseline />
          <Provider store={store}>{children}</Provider>
        </ThemeProvider>
      </StylesProvider>
    );
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
