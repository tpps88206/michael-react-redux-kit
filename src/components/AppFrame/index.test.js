import React from 'react';

import { render } from '@/utils/test-utils';
import AppFrame from './index';

it('renders AppFrame without crashing', () => {
  const div = document.createElement('div');
  render(<AppFrame />, div);
});

it('renders AppFrame with snapshot', () => {
  const { asFragment } = render(<AppFrame />);
  expect(asFragment()).toMatchSnapshot();
});
