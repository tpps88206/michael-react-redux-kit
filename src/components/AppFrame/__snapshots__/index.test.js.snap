// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AppFrame with snapshot 1`] = `
<DocumentFragment>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Github Repositories Searching
        </h6>
      </div>
    </header>
    <div />
  </div>
</DocumentFragment>
`;
